Source/WTF/wtf/MediaTime.cpp

```diff
@@ -184,11 +206,13 @@ MediaTime MediaTime::operator+(const MediaTime& rhs) const
     MediaTime a = *this;
     MediaTime b = rhs;
 
-    uint32_t commonTimeScale;
-    if (!leastCommonMultiple(a.m_timeScale, b.m_timeScale, commonTimeScale) || commonTimeScale > MaximumTimeScale)
-        commonTimeScale = MaximumTimeScale;
-    a.setTimeScale(commonTimeScale);
-    b.setTimeScale(commonTimeScale);
+    uint32_t commonTimeScale = a.m_timeScale;
+    if (a.m_timeScale != b.m_timeScale) {
+        if (!leastCommonMultiple(a.m_timeScale, b.m_timeScale, commonTimeScale) || commonTimeScale > MaximumTimeScale)
```

Comment:
Why adding the a ==b check here and in leastCommonMultiple?

Reply:
that made me look at `leastCommonMultiple` and fix a bug there !